rssfeed/tasks.py

```diff
@@ -0,0 +1,171 @@
+                        entry.link, attr)
+                    print(msg)
+        if hasattr(entry, "title"):
+            if entry.title == "":
+                if verbose:
+                    msg = 'rssfeed poll_feeds. Entry "%s" has a blank title' % (
+                        entry.link)
+                    print(msg)
+                continue
+        db_entry, created = Entry.objects.get_or_create(
+            feed=db_feed,
+            link=entry.link
+        )
+
+        if created:
+            if hasattr(entry, "published_parsed"):
+                published = datetime.fromtimestamp(
+                    mktime(entry.published_parsed)
+                )
+                db_entry.published = published
+            if hasattr(entry, "title"):
+                if len(entry.title) > MAX_LENGTH:
+                    db_entry.title = entry.title[0:MAX_LENGTH - 1]
+                else:
+                    db_entry.title = entry.title
+            # Mock does not support indexing. Verbose is set to True in test
+            # Different APIs have differently named keys for the media content
+            if hasattr(entry, "media_thumbnail"):
+                if len(entry.media_thumbnail[0]["url"]) > MAX_LENGTH:
+                    db_entry.image = \
+                        entry.media_thumbnail[0]["url"][0:MAX_LENGTH - 1]
+                else:
+                    db_entry.image = entry.media_thumbnail[0]["url"]
+            elif hasattr(entry, "media_context"):
+                if len(entry.media_context[0]["url"]) > MAX_LENGTH:
+                    db_entry.image = \
+                        entry.media_context[0]["url"][0:MAX_LENGTH - 1]
+                else:
+                    db_entry.image = entry.media_context[0]["url"]
+            elif hasattr(entry, "media_content"):
+                if len(entry.media_content[0]["url"]) > MAX_LENGTH:
+                    db_entry.image = \
+                        entry.media_content[0]["url"][0:MAX_LENGTH - 1]
+                else:
+                    db_entry.image = entry.media_content[0]["url"]
+            elif hasattr(entry, "summary") and not verbose:
+                if has_summary_image(entry):
+                    db_entry.image = find_article_image(entry.summary)
+                else:
+                    db_entry.image = ""
+            if hasattr(entry, "description"):
+                desc = BeautifulSoup.BeautifulSoup(entry.description).text
+                db_entry.description = desc
+            else:
+                db_entry = ""
+
+            db_entry.save()
+
+
+def has_summary_image(entry):
+    if entry.summary and (
+                    entry.summary.find(".jpg") or
+                    entry.summary.find(".gif") or
+                    entry.summary.find(".png")
+    ):
+        return True
+    else:
+        return False
+
+
+def find_article_image(summary):
+    if summary.find("jpg"):
+        image = summary[summary.find("http"):summary.find("jpg") + 3]
```

Comment:
It only finds the beginning index of the url, so for example if its `<p>http://www.example.com/image.jpg</p>` then it finds the index where `http` STARTS, which in this case is 3, and where the image url ends, which is like 30. So in the case of `https`, its irrelevant, since it only looks for the `h` in https.